Sources/ConcordiumSwiftSDK/Model/Account.swift

```diff
@@ -16,9 +49,373 @@ struct AccountAddress {
+                                $0.value
+                            },
+                            cmmIdCredSecSharingCoeff: v.commitments.idCredSecSharingCoeff.map {
+                                $0.value
+                            }
+                    )
+            )
+        }
+    }
+}
+
+typealias EncryptedAmount<C> = Data // is 'C' in Rust impl but the API returns Data
+
+struct AccountEncryptedAmount {
+    let selfAmount: EncryptedAmount<ArCurve>
+    let startIndex: UInt64
+    let aggregatedAmount: Data?
+    let incomingAmounts: [EncryptedAmount<ArCurve>]
+
+    static func fromGrpcType(_ grpc: Concordium_V2_EncryptedBalance) -> AccountEncryptedAmount {
+        AccountEncryptedAmount(
+                selfAmount: grpc.selfAmount.value,
+                startIndex: grpc.startIndex,
+                aggregatedAmount: grpc.aggregatedAmount.value,
+                incomingAmounts: grpc.incomingAmounts.map {
+                    $0.value
+                }
+        )
+    }
+}
+
+// TODO: Check out Java SDK for this...
+typealias BakerElectionVerifyKey = PublicKey
+typealias BakerSignatureVerifyKey = PublicKey
+typealias BakerAggregationVerifyKey = PublicKey
+
+typealias BakerId = AccountIndex
+
+struct BakerInfo {
+    let bakerId: BakerId
+    let bakerElectionVerifyKey: BakerElectionVerifyKey
+    let bakerSignatureVerifyKey: BakerSignatureVerifyKey
+    let bakerAggregationVerifyKey: BakerAggregationVerifyKey
+
+    static func fromGrpcType(_ grpc: Concordium_V2_BakerInfo) -> BakerInfo {
+        BakerInfo(
+                bakerId: grpc.bakerID.value,
+                bakerElectionVerifyKey: grpc.electionKey.value,
+                bakerSignatureVerifyKey: grpc.signatureKey.value,
+                bakerAggregationVerifyKey: grpc.aggregationKey.value
+        )
+    }
+}
+
+enum StakePendingChange {
+    case reduceStake(newStake: Amount, effectiveTime: Date)
+    case removeStake(effectiveTime: Date)
+
+    static func fromGrpcType(_ grpc: Concordium_V2_StakePendingChange) -> StakePendingChange? {
+        switch grpc.change {
+        case nil:
+            return nil
+        case let .reduce(r):
+            return .reduceStake(newStake: r.newStake.value, effectiveTime: dateFromUnixTimeMillis(r.effectiveTime.value))
+        case let .remove(r):
+            return .removeStake(effectiveTime: dateFromUnixTimeMillis(r.value))
+        }
+    }
+}
+
+struct AmountFraction {
+    let partsPerHundredThousand: UInt32
+
+    static func fromGrpcType(_ grpc: Concordium_V2_AmountFraction) -> AmountFraction {
+        AmountFraction(partsPerHundredThousand: grpc.partsPerHundredThousand)
+    }
+}
+
+enum OpenStatus: Int {
+    case openForAll = 0
+    case closedForNew = 1
+    case closedForAll = 2
+
+    static func fromGrpcType(_ grpc: Concordium_V2_OpenStatus) -> OpenStatus {
+        OpenStatus(rawValue: grpc.rawValue)!
+    }
+}
+
+struct CommissionRates {
+    let finalization: AmountFraction
```

Comment:
Do we need to make these fields optional (like in Rust?) or have some kind of indication whether the fields have been explicitly set (like in the generated Swift type)

Reply:
It's not, I'll give it another look.

Reply:
Fixed. I didn't notice this because of the retarded way the generated grpc types work. Nice catch!